props/saber_sa22c_buttons.h

```diff
@@ -0,0 +1,357 @@
+		  }
+		  SaberBase::DoBeginLockup();
+	  return true;
+		}
+
+        // Off functions
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_LONG, MODE_OFF):
+#if NUM_BUTTONS == 1
+	  if (!MODE_VOLUME);
+        next_preset();
+
+        return true;
+#endif
+#if NUM_BUTTONS > 1
+        StartOrStopTrack();
+#endif
+	return true;
+
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_LONG, MODE_OFF):
+      current_volume_ = dynamic_mixer.get_volume();
+      if (MODE_VOLUME){
+        MODE_VOLUME = false;
+        beeper.Beep(0.5, 3000);
+        STDOUT.println("Exit Volume Menu");
+      }
+      else{
+        MODE_VOLUME = true;
+        beeper.Beep(0.5, 3000);
+        STDOUT.println("Enter Volume Menu");
+      }
+      return true;
+
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF):
+        SaberBase::RequestMotion();
+	return true;
+
+      case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_OFF | BUTTON_POWER):
+	    if (!MODE_VOLUME) {
+          next_preset();
+		}
+
+	return true;
+
+#if NUM_BUTTONS < 3
+      case EVENTID(BUTTON_POWER, EVENT_HELD_LONG, MODE_OFF):
+#endif
+        if (!MODE_VOLUME) {
+          previous_preset();
+		}
+	return true;
+      case EVENTID(BUTTON_AUX, EVENT_HELD_LONG, MODE_OFF):
+      talkie.SayDigit((int)floorf(battery_monitor.battery()));
+      talkie.Say(spPOINT);
+      talkie.SayDigit(((int)floorf(battery_monitor.battery() * 10)) % 10);
+      talkie.SayDigit(((int)floorf(battery_monitor.battery() * 100)) % 10);
+      talkie.Say(spVOLTS);
+      return true;
+
+      case EVENTID(BUTTON_AUX2, EVENT_CLICK_SHORT, MODE_OFF):
+      STDOUT.println(NUM_BUTTONS);
+#if NUM_BUTTONS == 3
+        if (MODE_VOLUME){
+          STDOUT.println("Volume Down");
+          if (dynamic_mixer.get_volume() > (0.10 * VOLUME)){
+            current_volume_ = dynamic_mixer.get_volume();
+            current_volume_ -= (VOLUME * 0.10) ;
+            dynamic_mixer.set_volume(current_volume_);
+            beeper.Beep(0.5, 2000);
+            STDOUT.print("Current Volume: ");
+            STDOUT.println(dynamic_mixer.get_volume());
+          }
+          else{
+            beeper.Beep(0.5, 1000);
+          }
+        }
+
+#endif
+
+#ifdef DUAL_POWER_BUTTONS
+        if (!MODE_VOLUME) {
+			next_preset();
+
+		}
+#else
+        if (!MODE_VOLUME) {
+			previous_preset();
+
+		}
+#endif
+	return true;
+
+	// Events that needs to be handled regardless of what other buttons
+	// are pressed.
+      case EVENTID(BUTTON_POWER, EVENT_RELEASED, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_AUX, EVENT_RELEASED, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_AUX2, EVENT_RELEASED, MODE_ANY_BUTTON | MODE_ON):
+	if (SaberBase::Lockup()) {
+	  SaberBase::DoEndLockup();
+	  SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+	  return true;
+	}
+    }
+    return false;
+  }
+private:
+  bool aux_on_ = true;
+  bool pointing_down_ = false;
+  bool MODE_VOLUME = false;
+  int32_t current_volume_ = dynamic_mixer.get_volume();
```

Comment:
This seems weird.
If it's really the *current* volume, why do we need to keep it separately from dynamic_mixer.get_volume() ?

Reply:
This was probably the first code I wrote when i started tinkering with ProffieOS.  I'm sure it made sense at the time but, like yourself, I have no idea why i thought I needed to do this.  Removed.